### PR TITLE
Enhance openbmc framework with multi-processes

### PR DESCRIPTION
**Pull request by chenglch**
This patch integrate the process worker with the state machine
to support upload firmware in parallel.

partial-issue: #4055

### CODE REVIEW

File: xCAT-server/lib/xcat/plugins/openbmc.pm [hunk 1]
```diff
@@ -2257,10 +2244,57 @@ sub rflash_response {
...
+        if ($node_info{$node}{method} || $status_info{ $node_info{$node}{cur_status} }{method}) {
+            gen_send_request($node);
+        }
     } else {
         $wait_node_num--;
     }
     return;
 }
+
+sub rflash_upload {
```
Comment (zet809) [inline xCAT-server/lib/xcat/plugins/openbmc.pm]: I think we shall deal with the *state machine* in this subroutine. Both "RFLASH_FILE_UPLOAD_REQUEST" and "RFLASH_FILE_UPLOAD_RESPONSE".

Comment (chenglch) [reply xCAT-server/lib/xcat/plugins/openbmc.pm]: This is just a subroutine running in the child process as I'd like to split the subroutine into small parts especially with multi-processes. It's impossible to put the `state` transformation in the child process as the main process is responsible for maintaining of the state machine.  The change of the state in the child process could not reflect to the main process.

**Comment by zet809** [comment]
Looks great to me, I agree to merge, thx @chenglch

**Comment by whowutwut** [comment]
Also resolves: #4053

**Pull request merged**